### PR TITLE
Wordpress detection

### PR DESCRIPTION
**Pull request by ankk13**
Composer parser is modified to consider 2 types of file.
1. composer.lock
2. /wp-includes/version.php
This new file helps us to detect version of wordpress.

### CODE REVIEW

**Review by knqyf263 — COMMENTED** [review]
It should be outside of the composer analyzer.

File: analyzer/library/composer/composer.go [hunk 1]
```diff
@@ -19,21 +20,38 @@ func init() {
...
-		return nil, xerrors.Errorf("error with composer.lock: %w", err)
+	parsedLibs := []godepTypes.Library{}
+	var err error
+	r := bytes.NewReader(target.Content)
+	if strings.HasSuffix(target.FilePath, "/composer.lock") {
+		parsedLibs, err = composer.Parse(r)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse %s: %w", target.FilePath, err)
+		}
+	} else if strings.HasSuffix(target.FilePath, "/wp-includes/version.php") {
```
Comment (masahiro331) [inline analyzer/library/composer/composer.go:36]: IMO: If you want to use meaningful strings in multiple places, it is preferable to define them const.
like.
```
const (
    composer = "/composer.lock"
    wordpress = "/wp-includes/version.php"
)
```

File: analyzer/library/composer/composer.go [hunk 2]
```diff
@@ -19,21 +20,38 @@ func init() {
...
+		parsedLibs, err = composer.ParseWordPress(r)
+		if err != nil {
+			return nil, xerrors.Errorf("failed to parse %s: %w", target.FilePath, err)
+		}
 	}
-	return res, nil
+	if len(parsedLibs) == 0 {
```
Comment (masahiro331) [inline analyzer/library/composer/composer.go:42]: Other libraries return an AnalysisResult even if the detected library is empty.

File: analyzer/library/composer/composer_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	"testing"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/types"
+	godeptypes "github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_composer_Analyze(t *testing.T) {
```
Comment (masahiro331) [inline analyzer/library/composer/composer_test.go:14]: Please add more invalid case tests.

File: analyzer/library/composer/composer.go (hunk 2 above)
Comment (masahiro331) [reply analyzer/library/composer/composer.go:42]: Sorry, no problems skip this process. 
https://github.com/aquasecurity/fanal/blob/2cd043a8486e276a20c5607814e2972e0cbb3db4/analyzer/analyzer.go#L98-L100

**Comment by knqyf263** [comment]
We decided not to implement WordPress now.

**Pull request closed without merging**